api/v1/cluster_webhook.go

```diff
@@ -1079,6 +1079,29 @@ func (r *Cluster) validateConfiguration() field.ErrorList {
 		}
 	}
 
+	walLevel := postgres.WalLevelValue(sanitizedParameters[postgres.WalLevelParameter])
+	hasWalLevelRequirement := r.Spec.Instances > 1 || r.Spec.Backup.IsBarmanBackupConfigured() || r.IsReplica()
```

Comment:
In my opinion, instead of checking `r.Spec.Backup.IsBarmanBackupConfigured()` we need to simply check if `archive_mode` is `on`.